### ISSUE TITLE
Action Mailbox Conductor throws NoMethodError when creating inbound email

### ISSUE THREAD

**Issue by andrew2005** [rails/rails#43928]
### Steps to reproduce
- rails new conductor_bug
- cd conductor_bug
- rails action_mailbox:install
- rails db:migrate
- echo "class ApplicationMailbox < ActionMailbox::Base\n routing all: :catchall\nend" > 'app/mailboxes/application_mailbox.rb'
- echo "class CatchallMailbox < ApplicationMailbox\n def process; end\nend" > 'app/mailboxes/catchall_mailbox.rb'
- rails s
- Visit "http://localhost:3000/rails/conductor/action_mailbox/inbound_emails/new"
- Leave completely blank or attach a file (same result)
- Press Submit

Test file at bottom. 

### Expected behavior
Conductor should create inbound email

### Actual behavior
```
NoMethodError in Rails::Conductor::ActionMailbox::InboundEmailsController#create
undefined method `original_filename' for "":String

>> mail_params[:attachments]
=> [""]
```
Or if there is an attachment:
```
>> mail_params[:attachments]
=> ["", #<ActionDispatch::Http::UploadedFile:...
```

### Proposed Solution

mail_params[:attachments] is an array of attachments however the first element as a blank string whether there are attachments or not. This is causing the NoMethodError as the original_filename method is being called on the attachment elements. 

https://github.com/rails/rails/blob/260f735a1acf040fc0c306c3fbfc59e066539715/actionmailbox/app/controllers/rails/conductor/action_mailbox/inbound_emails_controller.rb#L25

The browsers I tested (Chrome, Edge, Brave, Safari) all pass this empty string into the attachments array. I think we need to handle this?

The above line could be changed to:

`mail_params[:attachments].reject(&:blank?).to_a.each do |attachments|`

Happy to open PR with below tests if solution is appropriate.

### System configuration
**Rails version**: 7.0.0

**Ruby version**: 3.0.2p107

### Script to reproduce
Note that all this does is add a blank string to the element array which is what is happening in practice. As above this might not be appropriate

```
# frozen_string_literal: true

require "bundler/inline"

gemfile(true) do
  source "https://rubygems.org"

  git_source(:github) { |repo| "https://github.com/#{repo}.git" }

  # Activate the gem you are reporting the issue against.
  gem "rails", "~> 7.0.0"
  gem "sqlite3"
  if RUBY_VERSION >= "3.1"
    # net-smtp, net-imap and net-pop were removed from default gems in Ruby 3.1, but is used by the `mail` gem.
    # So we need to add them as dependencies until `mail` is fixed: https://github.com/mikel/mail/pull/1439
    gem "net-smtp", require: false

    # digest gem, which is one of the default gems has bumped to 3.1.0.pre for ruby 3.1.0dev.
    # Also `net-smtp` v0.2.2 adds dependency to digest gem which attempts to install digest 3.0.0.
    gem "digest", "~> 3.1.0.pre", require: false
  end
end

require "active_record/railtie"
require "active_storage/engine"
require "action_mailbox/engine"
require "tmpdir"

class TestApp < Rails::Application
  config.root = __dir__
  config.hosts << "www.example.com"
  config.eager_load = false
  config.session_store :cookie_store, key: "cookie_store_key"
  secrets.secret_key_base = "secret_key_base"

  config.logger = Logger.new($stdout)
  Rails.logger  = config.logger

  config.active_storage.service = :local
  config.active_storage.service_configurations = {
    local: {
      root: Dir.tmpdir,
      service: "Disk"
    }
  }

  config.action_mailbox.ingress = :relay
end

ENV["DATABASE_URL"] = "sqlite3::memory:"

Rails.application.initialize!

require ActiveStorage::Engine.root.join("db/migrate/20170806125915_create_active_storage_tables.rb").to_s
require ActionMailbox::Engine.root.join("db/migrate/20180917164000_create_action_mailbox_tables.rb").to_s

ActiveRecord::Schema.define do
  CreateActiveStorageTables.new.change
  CreateActionMailboxTables.new.change
end

class ApplicationMailbox < ActionMailbox::Base
  routing (/^replies@/i) => :replies
end

class RepliesMailbox < ActionMailbox::Base
  def process
    $processed = mail.subject
  end
end

require "minitest/autorun"

File.write('test_attachment.txt','test')

class InboundEmailsControllerTestNew < ActionDispatch::IntegrationTest
  test "create inbound email with attachment" do
    with_rails_env("development") do
      post rails_conductor_inbound_emails_path, params: {
        mail: {
          from: "Jason Fried <jason@37signals.com>",
          to: "Replies <replies@example.com>",
          subject: "Let's debate some attachments",
          body: "Let's talk about these images:",
          attachments: [ fixture_file_upload("test_attachment.txt") ]
        }
      }

      assert_response :redirect
    end
  end

  test "create inbound email with attachment and empty string" do
    with_rails_env("development") do
      post rails_conductor_inbound_emails_path, params: {
        mail: {
          from: "Jason Fried <jason@37signals.com>",
          to: "Replies <replies@example.com>",
          subject: "Let's debate some attachments",
          body: "Let's talk about these images:",
          attachments: [ "", fixture_file_upload("test_attachment.txt") ]
        }
      }

      assert_response :redirect
    end
  end

  test "create inbound email with no attachments" do
    with_rails_env("development") do
      post rails_conductor_inbound_emails_path, params: {
        mail: {
          from: "Jason Fried <jason@37signals.com>",
          to: "Replies <replies@example.com>",
          subject: "Let's debate some attachments",
          body: "Let's talk about these images:",
          attachments: [  ]
        }
      }

      assert_response :redirect
    end
  end

  test "create inbound email with no attachments and blank string" do
    with_rails_env("development") do
      post rails_conductor_inbound_emails_path, params: {
        mail: {
          from: "Jason Fried <jason@37signals.com>",
          to: "Replies <replies@example.com>",
          subject: "Let's debate some attachments",
          body: "Let's talk about these images:",
          attachments: [ "" ]
        }
      }

      assert_response :redirect
    end
  end

  private
    def with_rails_env(env)
      old_rails_env = Rails.env
      Rails.env = env
      yield
    ensure
      Rails.env = old_rails_env
    end
end
```

**Comment by jakehockey10** [comment]
I'm also experiencing this issue with
- Rails 7.0.1
- ruby 3.0.3p157

**Comment by AlexanderCohen** [comment]
I'm also experiencing this issue. Rails 7.0.1 Ruby 3.0.3p157

**Comment by brudermusscode** [comment]
Experiencing it as well, on **3.1.0p0@7.0.1**

**Comment by armiiller** [comment]
And for those of you looking for a quick fix, add a monkey patch in `config\application.rb`

```ruby
...
config.to_prepare do 
      
  # some monkey patches
  Rails::Conductor::ActionMailbox::InboundEmailsController.class_eval do
    private
    def new_mail
      Mail.new(mail_params.except(:attachments).to_h).tap do |mail|
        mail[:bcc]&.include_in_headers = true
        mail_params[:attachments].to_a.compact_blank.each do |attachment|
          mail.add_file(filename: attachment.original_filename, content: attachment.read)
        end
      end
    end
  end    
end
...
```

The `compact_blank` fixes this temporarily

**Comment by ericsaupe** [comment]
I think this was fixed by #45103

**Comment by Merovex** [comment]
I'm having this problem as of 7.0.3
